action/penalization/allegation.go

```diff
@@ -0,0 +1,133 @@
+package penalization
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	"github.com/tendermint/tendermint/libs/kv"
+
+	"github.com/Oneledger/protocol/action"
+	gov "github.com/Oneledger/protocol/data/governance"
+	"github.com/Oneledger/protocol/data/keys"
+)
+
+var _ action.Msg = &Allegation{}
+
+type Allegation struct {
+	ValidatorAddress keys.Address
+	MaliciousAddress keys.Address
+	BlockHeight      int64
+	ProofMsg         string
+}
+
+func (r Allegation) Marshal() ([]byte, error) {
+	return json.Marshal(r)
+}
+
+func (r *Allegation) Unmarshal(data []byte) error {
+	return json.Unmarshal(data, r)
+}
+
+func (r Allegation) Signers() []action.Address {
+	return []action.Address{r.ValidatorAddress.Bytes()}
+}
+
+func (r Allegation) Type() action.Type {
+	return action.ALLEGATION
+}
+
+func (r Allegation) Tags() kv.Pairs {
+	tags := make([]kv.Pair, 0)
+
+	tag := kv.Pair{
+		Key:   []byte("tx.type"),
+		Value: []byte(r.Type().String()),
+	}
+	tag2 := kv.Pair{
+		Key:   []byte("tx.validator"),
+		Value: r.ValidatorAddress.Bytes(),
+	}
+
+	tags = append(tags, tag, tag2)
```

Comment:
Could you add MaliciousAddress, height and proof to tags? Then explorer will be able to catch full information.